btree/palm/tree_test.go

```diff
@@ -292,18 +291,39 @@ func BenchmarkReadAndWrites(b *testing.B) {
 		keys = append(keys, generateRandomKeys(numItems))
 	}
 
-	tree := newTree(16)
-	var wg sync.WaitGroup
-	wg.Add(b.N)
+	tree := newTree(16, 16)
 	b.ResetTimer()
 
 	for i := 0; i < b.N; i++ {
 		tree.Insert(keys[i]...)
 		tree.Get(keys[i]...)
-		wg.Done()
 	}
+}
 
-	wg.Wait()
+func BenchmarkSimultaneousReadsAndWrites(b *testing.B) {
+	numItems := 1000
+	numRoutines := 8
+	keys := make([]Keys, 0, numRoutines)
+	for i := 0; i < numRoutines; i++ {
+		keys = append(keys, generateRandomKeys(numItems))
+	}
+
+	tree := newTree(16, 16)
+	var wg sync.WaitGroup
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		wg.Add(numRoutines)
+		for j := 0; j < numRoutines; j++ {
+			go func(j int) {
+				tree.Insert(keys[j]...)
+				tree.Get(keys[j]...)
+				wg.Done()
+			}(j)
+		}
+
+		wg.Wait()
+	}
```

Comment:
Shouldn't the tree be reset at the beginning of every iteration of this benchmark?